lib/new_relic/agent/llm.rb

```diff
@@ -3,9 +3,47 @@
 # frozen_string_literal: true
 
 require_relative 'llm/llm_event'
-require_relative 'llm/chat_completion'
 require_relative 'llm/chat_completion_message'
 require_relative 'llm/chat_completion_summary'
 require_relative 'llm/embedding'
-require_relative 'llm/feedback'
 require_relative 'llm/response_headers'
+
+module NewRelic
+  module Agent
+    class LLM
+      INPUT = 'input'
+      CONTENT = 'content'
+      SEGMENT_PATTERN = %r{Llm/.*/OpenAI/.*}.freeze
+
+      def self.instrumentation_enabled?
+        NewRelic::Agent.config[:'ai_monitoring.enabled']
+      end
+
+      # LLM content-related attributes are exempt from the 4095 byte limit
+      def self.exempt_event_attribute?(type, key)
+        return false unless instrumentation_enabled?
+
+        (type == NewRelic::Agent::Llm::Embedding::EVENT_NAME && key == INPUT) ||
+          (type == NewRelic::Agent::Llm::ChatCompletionMessage::EVENT_NAME && key == CONTENT)
```

Comment:
I was originally worried about `type` being a string. One concern is that a symbol might appear which would fail the `==` check, and another concern is that for performance the use of a symbol might actually be preferred.

From what I can tell, `CustomEventAggregator#record` takes care of forcing `type` to become a string by routing it through a `@type_strings` lookup. We can evaluate a switch to symbols for performance post GA.

Reply:
Added a new story for this: https://github.com/newrelic/newrelic-ruby-agent/issues/2520